grails-app/domain/org/pih/warehouse/order/Order.groovy

```diff
@@ -456,8 +456,24 @@ class Order implements Serializable {
         return !hasInvoice && isPrepaymentRequired && hasActiveItemsOrAdjustments
     }
 
+    /*
+        Should return true if at least one PO item is invoiceable.
+        A PO item is invoiceable if the order has a prepayment invoice and at least one of those requirements are fulfilled:
+            1. There is an order adjustment on that item that hasn’t already been invoiced
+            2. The item’s shipment has been cancelled and that cancellation hasn’t already been invoiced
+            3. Quantity greater than 0 of the item has been shipped and not all of that quantity has already been invoiced.
+    */
     Boolean getCanGenerateInvoice() {
-        return hasPrepaymentInvoice && isShipped() && !hasRegularInvoice
+        // If prepayment invoice doesn't exist we can't generate invoice
+        if (!hasPrepaymentInvoice) {
+            return false
+        }
+
+        return orderItems.any {
+            it.hasNotInvoicedAdjustment || // condition related to first point
```

Comment:
What if the order adjustment is not tied to any order item? This is possible too